main.js

```diff
@@ -884,59 +961,76 @@ function extendObjects(tasks, callback) {
 }
 
 function setMeta() {
-    const id = 'system.host.' + hostname;
+    const id = hostObjectPrefix;
 
     objects.getObject(id, (err, oldObj) => {
-        const newObj = {
-            _id:  id,
-            type: 'host',
-            common: {
-                name:             hostname,
-                title:            oldObj && oldObj.common && oldObj.common.title ? oldObj.common.title : ioPackage.common.title,
-                installedVersion: version,
-                platform:         ioPackage.common.platform,
-                cmd:              process.argv[0] + ' ' + (process.execArgv.join(' ') + ' ').replace(/--inspect-brk=\d+ /, '') + process.argv.slice(1).join(' '),
-                hostname:         hostname,
-                address:          getIPs(),
-                type:             ioPackage.common.name
-            },
-            native: {
-                process: {
-                    title:      process.title,
-                    versions:   process.versions,
-                    env:        process.env
+        let newObj;
+        if (compactGroupController) {
+            newObj = {
+                _id: id,
+                type: 'device',
+                common: {
+                    name: hostname + '.compactgroup' + compactGroup,
+                    cmd: process.argv[0] + ' ' + (process.execArgv.join(' ') + ' ').replace(/--inspect-brk=\d+ /, '') + process.argv.slice(1).join(' '),
```

Comment:
Isn't this basically just the command line with `--inspect-brk` filtered out? Looks pretty complicated.

Reply:
i did not touched this line ;-)
In fact process.execArgv is special because these in there are not in regular argv ... so they are added and then the one is removed